### PR TITLE
upgrade oc client to v4

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-2684

### CODE REVIEW

**Comment by maorfr** [comment]
this is being tested using the `make rc` command in https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/11374

File: dockerfiles/Dockerfile [hunk 1]
```diff
@@ -33,10 +33,10 @@ ENV GIT_SECRETS_VERSION=1.3.0
 ENV JSONNET_VENDOR_DIR=/opt/jsonnet-bundler/vendor
 
 RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && \
-    curl -L https://github.com/openshift/origin/releases/download/v3.11.0/openshift-origin-client-tools-v3.11.0-0cbc58b-linux-64bit.tar.gz | tar -xvz && \
-    pushd openshift-origin-client* && \
+    curl https://mirror.openshift.com/pub/openshift-v4/clients/ocp/latest/openshift-client-linux.tar.gz -o oc.tar.gz &&\
```
Comment (apahim) [inline dockerfiles/Dockerfile:36]: We should probably pin-point a version from https://mirror.openshift.com/pub/openshift-v4/clients/ocp/ so we always know what we are running, otherwise any compatibility break from latest will blow in the hands of someone doing unrelated changes.

**Review by apahim — APPROVED** [review]
LGTM. Not sure we should go with latest though, but I'd be ok with it.

**Comment by maorfr** [comment]
latest is greatest! pinned to 4.6.1, which is what we are trying to get to for our clusters ATM.

File: dockerfiles/Dockerfile (hunk 1 above)
Comment (maorfr) [reply dockerfiles/Dockerfile]: pinned.

**Comment by rporres** [comment]
Late to the party. A couple of things I don't particularly like:

* A newer client going against older clusters. It can blow in the least expected moment.
* A solution to this different than matching cluster version with client version.

**Comment by maorfr** [comment]
we decided to just go with this approach since it appears that most of us are running oc v4 locally anyway, and no one has seen anything funky. we also only have 2-3 remaining v3 clusters.

**Comment by rporres** [comment]
> no one has seen anything funky.

famous last words

**Comment by rporres** [comment]
https://issues.redhat.com/browse/APPSRE-2724 :)
